071-kafkabrige-consumer-producer.md

```diff
@@ -0,0 +1,85 @@
+# Enhance KafkaBridge resource with consumer inactivity timeout and HTTP consumer/producer parts enablement
+
+Providing support in the Strimzi Kubernetes Operator for following properties supported in the Strimzi HTTP Kafka Bridge:
+* http.consumer.timeoutSeconds   - For deleting inactive consumers after a timeout (disabled by default).
+* http.consumer.enabled          - To enable/disable the HTTP consumer part (enabled by default).
+* http.producer.enabled          - To enable/disable the HTTP producer part (enabled by default).
+
+## Current situation
+
+Properties are not yet supported in the Strimzi Kubernetes Operator.
+
+## Motivation
+
+Raised in the discussion here: https://github.com/strimzi/strimzi-kafka-operator/issues/8732 and triaged on 29.6.2023: We should enable the configuration of these fields. But how should the API look like? Should have a proposal to clarify the API changes.
+
+## Proposal
+
+This proposal suggests based on discussions here: https://github.com/strimzi/strimzi-kafka-operator/pull/9820 to add the http enablement for the consumer and producer in their respective sessions, i.e. spec.consumer.enabled and spec.producer.enabled, as well as adding timeoutSeconds to the spec.consumer properties.
+
+Suggestion:
+
+```yaml
+apiVersion: "kafka.strimzi.io/v1beta2"
+kind: "KafkaBridge"
+metadata:
+  name: "test-kafka-bridge"
+spec:
+  replicas: 1
+  image: "my-test-image"
+  bootstrapServers: "my-cluster-kafka:9092"
+  consumer:
+    enabled: true
+    timeoutSeconds: 60
+    config:
+      foo: "bar"
+  producer:
+    enabled: false
+    config:
+      foo: "buz"
+  enableMetrics: false
+```
+
+## Affected/not affected projects
+
+Strimzi Kafka Bridge - PR raised for the HTTP Bridge to use the new ENV parameters introduced from the operator: https://github.com/strimzi/strimzi-kafka-bridge/pull/882
+
+## Compatibility
+
+http.consumer.enabled and http.producer.enabled both defaults to true, thus there are no impacts on backward compatibility.
+
+If http.consumer.timeoutSeconds is not specified, it should default to -1 to have the same effect, as when it is omitted from the application.properties.
```

Comment:
It seems you are referring to the property you would have in the `KafkaBridge` which is omitted but it's not `http.consumer.timeoutSeconds` but it's `spec.consumer.timeoutSeconds`.

```suggestion
Not specifying `spec.consumer.timeoutSeconds` implies not configuring the corresponding bridge parameter `http.consumer.timeoutSeconds` which defaults to -1 to have the same effect (no timeout), thus there are no impacts on backward compatibility.
```